integtests/apitester/apitester.go

```diff
@@ -50,57 +43,88 @@ func (c *Config) Run() error {
 	return nil
 }
 
-func (c *Config) check(testCase Case) error {
-	controller, err := broker.New()
-	ts := httptest.NewServer(http.HandlerFunc(controller.CreateServiceInstance))
-	defer ts.Close()
-
-	if testCase.Endpoint == datafactory.GetGetPXCInstanceData().Endpoint {
-		return c.checkStatus(ts, testCase)
+func (c *Config) check(testCase structs.CaseData) error {
+	// Here we waitning
+	if testCase.ReqType == "GET" && testCase.RespStatus == 200 {
+		return c.waitForSucceed(testCase)
 	}
-	respData, respStatus, err := c.Request(ts.URL, testCase.ReqType, testCase.ReqData)
+	respData, respStatus, err := c.Request(c.Address+testCase.Endpoint, testCase.ReqType, testCase.ReqData)
 	if err != nil {
 		return err
 	}
-	if respStatus != http.StatusOK || respStatus != http.StatusAccepted {
-		log.Println(respStatus)
+	if respStatus != testCase.RespStatus {
 		return fmt.Errorf("Wrong status")
 	}
-	log.Println(respStatus, string(respData))
-	if len(testCase.ReqData) > 0 && string(respData) != testCase.RespData {
-		return fmt.Errorf("Wrong responce")
-	}
-	return nil
+
+	return checkRespData(testCase.RespData, respData)
 }
 
-func (c *Config) checkStatus(ts *httptest.Server, testCase Case) error {
+type GetInstanceResp struct {
+	LastOperation struct {
+		State       string `json:"state"`
+		Description string `json:"description"`
+	} `json:"last_operation"`
+}
+
+func (c *Config) waitForSucceed(testCase structs.CaseData) error {
 	startTime := time.Now()
-	endTime := startTime.Local().Add(time.Second * time.Duration(180))
+	endTime := startTime.Add(time.Second * 250)
 	ticker := time.NewTicker(2 * time.Second)
-
+	defer ticker.Stop()
 	var instResp GetInstanceResp
 	for t := range ticker.C {
-		respData, _, err := c.Request(ts.URL, testCase.ReqType, testCase.ReqData)
+		respData, status, err := c.Request(c.Address+testCase.Endpoint, testCase.ReqType, testCase.ReqData)
 		if err != nil {
 			return err
 		}
 		err = json.Unmarshal(respData, &instResp)
 		if err != nil {
-			return err
+			log.Println(err)
+			continue
 		}
-		if instResp.State != "succeeded" {
-			return nil
+
+		if instResp.LastOperation.State == "succeeded" {
+			fmt.Println() // just for create new line after ticker
+			if status != testCase.RespStatus {
+				return errors.New("Wrong resp status. Have '" + instResp.LastOperation.State + "', want 'succeeded'")
```

Comment:
1. "Error strings should not be capitalized"
2. It is definitely something wrong here. You check `status` and `testCase.RespStatus`, but the error contain quite a bit different statements.